test/extended/dns/dns.go

```diff
@@ -492,7 +492,13 @@ var _ = Describe("[sig-network-edge] DNS", func() {
 
 		// Run a pod which probes DNS and exposes the results by HTTP.
 		By("creating a pod to probe DNS")
-		pod := createDNSPod(f.Namespace.Name, cmd)
+		controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		nodeSelector := map[string]string{}
+		if *controlPlaneTopology != ocpv1.ExternalTopologyMode {
+			nodeSelector = map[string]string{"node-role.kubernetes.io/master": ""}
+		}
```

Comment:
If you move this code block up to L414, you don't need to declare it again later.